src/serena/agent.py

```diff
@@ -79,23 +83,31 @@ def __init__(self) -> None:
         if not os.path.exists(config_file):
             raise FileNotFoundError(f"Serena configuration file not found: {config_file}")
         with open(config_file, encoding="utf-8") as f:
-            config_yaml = yaml.safe_load(f)
+            try:
+                log.info(f"Loading Serena configuration from {config_file}")
+                config_yaml = yaml.safe_load(f)
+            except Exception as e:
+                raise ValueError(f"Error loading Serena configuration from {config_file}: {e}") from e
 
         # read projects
         self.projects: dict[str, ProjectConfig] = {}
-        for project_config_path in config_yaml["projects"]:
+        for project_config_path in config_yaml.get("projects", []):
             project_config_path = Path(project_config_path)
             if not project_config_path.is_absolute():
                 project_config_path = Path(serena_root_path()) / project_config_path
             if project_config_path.is_dir():  # assume project file in default location
                 project_config_path = project_config_path / ProjectConfig.SERENA_MANAGED_DIR / ProjectConfig.SERENA_DEFAULT_PROJECT_FILE
             if not project_config_path.is_file():
                 raise FileNotFoundError(f"Project file not found: {project_config_path}")
+            log.info(f"Loading project configuration from {project_config_path}")
             project_config = ProjectConfig.from_yml(project_config_path)
-            self.projects[project_config.project_name] = project_config
+            try:
+                self.projects[project_config.project_name] = project_config
+            except Exception as e:
+                raise ValueError(f"Error loading project configuration from {project_config_path}: {e}") from e
```

Comment:
This makes no sense to me. Did you mean to include the `from_yml` call?

Reply:
Yeah, out of place, sry